### ISSUE TITLE
What semantics do we want for the MeTTa import op?

### ISSUE THREAD

**Issue by luketpeterson** [trueagi-io/hyperon-experimental#509]
From the implementation of `ImportOp`

"Import" can mean several (3) different things.  In Python parlance, it can mean
1. "import module" opt. ("as bar")
2. "from module import foo" opt. ("as bar")
3. "from module import *"

Do we want one MeTTa operation with multiple ways of invoking it?  Or do we want different
implementations for different versions of the import operation?  (since we don't have key-words)
like "from" and "as" (unless we want to add them)

The old version of this operation supported 1. or 3., depending on whether a "space" argument
mapped to an atom that already existed or not.  If the space atom existed and was a Space, then
the operation would perform behavior 3.  Otherwise it would perform behavior 1.

For now, I have only implemented behavior 1 in this operation.
Underlying functionality for behavior 2 exists in MettaMod::import_item_from_dependency_as, but it isn't called yet because I wanted to discuss the way to expose it as a MeTTa op.
For behavior 3, there are deeper questions about desired behavior around transitive imports, Tokenizer entries, etc.  I have summarized those concerns in the discussion comments above MettaMod::import_all_from_dependency

**Comment by luketpeterson** [comment]
I've come to feel somewhat strongly that `import` should be renamed to `use` to avoid the various (and often contradicting) expectations voiced by many people about `import` is supposed to work.

**Comment by vsbogd** [comment]
> I've come to feel somewhat strongly that import should be renamed to use to avoid the various (and often contradicting) expectations voiced by many people about import is supposed to work.

I don't have strong opinion here except we need accurately migrate the code. I also feel that main issue is semantics of the `import` is not explained well and fixed. For instance nobody expects from `import` in Python that it behaves as `include`. So we need to fix and clearly explain these semantics on the first place. Renaming `import` to `use` is a second priority and we may do this to if @Necr0x0Der doesn't have objections.

**Comment by luketpeterson** [comment]
I agree it makes no sense to do it now.  But when we redo the arguments, I believe the name should change as well.

**Comment by Necr0x0Der** [comment]
If we change the signature of `import` in such a way that it will be incompatible with the existing code, then renaming will be indeed a good idea to reduce confusion.
